backend/src/common/plugins/authorization/authorization.plugin.ts

```diff
@@ -0,0 +1,68 @@
+import { type FastifyInstance, type FastifyPluginCallback } from 'fastify';
+import fp from 'fastify-plugin';
+
+import { type UserFindResponseDto } from '~/bundles/users/types/types.js';
+import { type UserService } from '~/bundles/users/users.js';
+import { ControllerHooks } from '~/common/controller/controller.js';
+import { HttpCode } from '~/common/http/http.js';
+import { tokenService } from '~/common/services/services.js';
+
+type AuthOptions = {
+    services: {
+        userService: UserService;
+    };
+    routesWhiteList: string[];
+};
+
+declare module 'fastify' {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+    interface FastifyRequest {
+        user: UserFindResponseDto;
+    }
+}
+
+const authorizationPlugin: FastifyPluginCallback<AuthOptions> = (
+    fastify: FastifyInstance,
+    { services, routesWhiteList }: AuthOptions,
+    done,
+) => {
+    fastify.decorateRequest('user', null);
+
+    fastify.addHook(ControllerHooks.ON_REQUEST, async (request, reply) => {
+        try {
+            const isWhiteRoute = routesWhiteList.includes(request.routerPath);
+
+            if (isWhiteRoute) {
+                return;
+            }
+
+            if (request.headers.authorization) {
```

Comment:
Get token, and if we do not have the token throw an error

Reply:
Hello, I am not sure what this means. Isn't it already handled in the current code?

```
if (request.headers.authorization) {
        // code here
} else {
    throw new Error(AuthorizationErrorMessages.NOT_AUTHORIZED);
}

```

Reply:
Current iteration:
![image](https://github.com/BinaryStudioAcademy/bsa-2023-bsa-talents/assets/90120572/36dad134-c74b-44e4-baee-0142886ffa45)

Reply:
I don't like `if` block in the `if` block.

```js
    const [, token] = request.headers.authorization?.split(' ') ?? [];

    if (!token) {
        throw error;
    }

    const authorizedUser;

    if (!authorizedUser) {
        throw error;
    }
```